File: Problems_and_Solutions/days-10 to 19/day-13/Java/Solution2.java
import java.util.HashMap;
import java.util.Arrays;

class Solution2 {
    public static int[] twoSum1(int[] nums, int target) {
        for(int i=0; i<nums.length; i++){
            for(int j=i+1; j<nums.length; j++){
                if(nums[j] == target-nums[i]){
                    return new int[] {i, j};
                }
            }
        }
        return null;
    }

    public static int[] twoSum2(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i=0; i<nums.length; i++) {
            int complement = target - nums[i];
            if (map.containsKey(complement)) {
                return new int[]{map.get(complement), i};
            }
            map.put(nums[i], i);
        }
        return null;
    }

    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        int target = 9;

        int[] res = twoSum1(nums, target);
        System.out.println(Arrays.toString(res));
    }
}

File: Problems_and_Solutions/days-10 to 19/day-13/c++/solution 2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        int i=0,j=0,temp=0;
        for(i=0;i<n-1;i++)
        {
            temp=target-nums[i];
            for(j=i+1;j<n;j++)
            {
                if(temp==nums[j])
                    return vector<int>{i,j};
            }
        }
        return vector<int>{0,0};
    }
   
};